qdev_wrappers/dataset/doNd.py

```diff
@@ -141,15 +142,21 @@ def do1d(
             called before the measurements start
         exit_actions: A list of functions taking no arguments that will be
             called after the measurements ends
+        register_params: A list of setpoint parameters to be registered in the
+            measurement but not scanned.
         do_plot: should png and pdf versions of the images be saved after the
             run.
 
     Returns:
         The run_id of the DataSet created
     """
     meas = Measurement()
-    _register_parameters(meas, (param_set,))
-    _register_parameters(meas, param_meas, setpoints=(param_set,))
+    if register_params is None:
+        register_params = tuple()
+    setpoint_params = (param_set,) + tuple(register_params)
+    _register_parameters(meas, setpoint_params)
+    _register_parameters(meas, param_meas, setpoints=setpoint_params)
+    params_to_measure = param_meas + tuple(register_params)
```

Comment:
@emartinez Could you elaborate the reasoning to add `register_params` as both set points and also parameters to be measured? Are there any down falls of this approach? Do we have any performance degradation?